vm/devices/net/netvsp/src/lib.rs

```diff
@@ -1175,25 +1191,39 @@ impl VmbusDevice for Nic {
         open_request: &OpenRequest,
     ) -> Result<(), ChannelOpenError> {
         // Start the coordinator task if this is the primary channel.
-        let state = if channel_idx == 0 {
+        let (state, packet_filter) = if channel_idx == 0 {
             self.insert_coordinator(1, false);
-            WorkerState::Init(None)
+            // No rx traffic on primary channel until guest sets the packet filter.
+            (WorkerState::Init(None), rndisprot::NDIS_PACKET_TYPE_NONE)
         } else {
             self.coordinator.stop().await;
             // Get the buffers created when the primary channel was opened.
             let buffers = self.coordinator.state().unwrap().buffers.clone().unwrap();
-            WorkerState::Ready(ReadyState {
-                state: ActiveState::new(None, buffers.recv_buffer.count),
-                buffers,
-                data: ProcessingData::new(),
-            })
+            // Get the packet filter of the primary channel.
+            self.coordinator.state_mut().unwrap().workers[0]
```

Comment:
Instead of getting this out of the primary channel, can we have the primary channel report the new packet filter to the coordinator and so instead get it out of the coordinator?